packages/client/src/object/fetchPage.ts

```diff
@@ -83,6 +112,68 @@ export type FetchPageResult<
   >
 >;
 
+function objectSetToSearchJsonV2(
+  objectSet: ObjectSet,
+  expectedApiName: string,
+  existingWhere: SearchJsonQueryV2 | undefined = undefined,
+): SearchJsonQueryV2 | undefined {
+  if (objectSet.type === "base") {
+    if (objectSet.objectType !== expectedApiName) {
+      throw new Error(
+        `Expected objectSet.objectType to be ${expectedApiName}, but got ${objectSet.objectType}`,
+      );
+    }
+
+    return existingWhere;
+  }
+
+  if (objectSet.type === "filter") {
+    return objectSetToSearchJsonV2(
+      objectSet.objectSet,
+      expectedApiName,
+      objectSet.where,
+    );
+  }
+
+  throw new Error(`Unsupported objectSet type: ${objectSet.type}`);
```

Comment:
Does this work if you have like `client(iface).where().where().fetchPage()`? Not sure if the MinimalObjectSet removes the ability to make something more complicated.

Reply:
Added an example to confirm it works